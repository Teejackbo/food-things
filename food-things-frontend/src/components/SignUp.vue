<template>
  <Spinner v-if="submitting" :fullScreen="true" />
  <div v-else>
    <h1>Sign Up</h1>
    <form @submit.prevent="handleSubmit">
      <div>
        <InputField
          v-model="registrationInfo.first_name"
          display="inline-block"
          inputWidth="80px"
          type="text"
          placeholder="Joseph"
          label="First Name"
          name="first_name"
        />

        <InputField
          v-model="registrationInfo.last_name"
          display="inline-block"
          inputWidth="100px"
          type="text"
          placeholder="Bloggs"
          label="Last Name"
          name="last_name"
        />

        <InputField
          v-model="registrationInfo.email"
          display="inline-block"
          inputWidth="218px"
          type="email"
          placeholder="joebloggs@example.com"
          label="Email Address"
          name="email"
        />
      </div>

      <div>
        <InputField
          v-model="registrationInfo.username"
          display="inline-block"
          type="text"
          placeholder="joe_bloggs"
          label="Username"
          name="username"
        />

        <InputField
          v-model="registrationInfo.password"
          display="inline-block"
          inputWidth="240px"
          type="password"
          placeholder="Must have at least 5 characters."
          label="Password"
          name="password"
        />
      </div>
      <PrimaryButton type="success">Sign Up</PrimaryButton>
    </form>
  </div>
</template>

<script>
import { mapActions, mapState } from 'vuex'
import isEmpty from 'lodash/isEmpty'
import { REGISTER_USER_ACTION } from '@/store/modules/user/types'

export default {
  name: 'SignUp',

  data: () => ({
    registrationInfo: {
      first_name: '',
      last_name: '',
      email: '',
      username: '',
      password: '',
    },
    submitting: false,
  }),

  computed: mapState({
    user: state => state.user.user,
  }),

  methods: {
    ...mapActions({
      register: REGISTER_USER_ACTION,
    }),

    async handleSubmit() {
      this.submitting = true
      await this.register(this.registrationInfo)
      this.submitting = false

      if (!isEmpty(this.user)) this.$router.push({ name: 'home' })
    },
  },
}
</script>
