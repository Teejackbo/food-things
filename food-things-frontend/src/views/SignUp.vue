<template>
  <Spinner v-if="submitting" :fullScreen="true" />
  <div v-else>
    <h1>Sign Up</h1>
    <form @submit.prevent="handleSubmit">
      <div>
        <InputField
          v-model="registrationInfo.first_name"
          display="inline-block"
          inputWidth="80px"
          type="text"
          placeholder="Joseph"
          name="first_name"
        >
          <label for="first_name">First Name</label>
        </InputField>

        <InputField
          v-model="registrationInfo.last_name"
          display="inline-block"
          inputWidth="100px"
          type="text"
          placeholder="Bloggs"
          name="last_name"
        >
          <label for="last_name">Last Name</label>
        </InputField>

        <InputField
          v-model="registrationInfo.email"
          display="inline-block"
          inputWidth="218px"
          type="text"
          placeholder="joebloggs@example.com"
          name="email"
        >
          <label for="email">Email Address</label>
        </InputField>
      </div>

      <div>
        <InputField
          v-model="registrationInfo.username"
          display="inline-block"
          type="text"
          placeholder="joe_bloggs"
          name="username"
        >
          <label for="username">Username</label>
        </InputField>

        <InputField
          v-model="registrationInfo.password"
          display="inline-block"
          inputWidth="240px"
          type="password"
          placeholder="Must have at least 5 characters."
          name="password"
        >
          <label for="password">Password</label>
        </InputField>
      </div>
      <PrimaryButton type="success">Sign Up</PrimaryButton>
    </form>
  </div>
</template>

<script>
import { mapActions, mapState } from 'vuex'
import isEmpty from 'lodash/isEmpty'
import { REGISTER_USER_ACTION } from '@/store/modules/user/types'

export default {
  name: 'SignUp',

  data: () => ({
    registrationInfo: {
      first_name: '',
      last_name: '',
      email: '',
      username: '',
      password: '',
    },
    submitting: false,
  }),

  computed: mapState({
    user: state => state.user.user,
  }),

  methods: {
    ...mapActions({
      register: REGISTER_USER_ACTION,
    }),

    async handleSubmit() {
      this.submitting = true
      await this.register(this.registrationInfo)
      this.submitting = false

      if (!isEmpty(this.user)) this.$router.push({ name: 'home' })
    },
  },
}
</script>
