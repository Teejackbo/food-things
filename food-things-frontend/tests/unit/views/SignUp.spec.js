import { shallowMount } from '@vue/test-utils'
import SignUp from '@/views/SignUp'
import registerGlobals from '../components/global/'

registerGlobals()

describe('SignUp', () => {
  it('Should match snapshot.', () => {
    const wrapper = shallowMount(SignUp)

    expect(wrapper.element).toMatchSnapshot()
  })

  it('Should render a fullscreen spinner if the component is submitting.', () => {
    const wrapper = shallowMount(SignUp, {
      data() {
        return {
          submitting: true,
        }
      },
    })

    expect(wrapper.attributes()).toEqual({ fullscreen: 'true' })
    expect(wrapper.element).toMatchSnapshot()
  })
})
