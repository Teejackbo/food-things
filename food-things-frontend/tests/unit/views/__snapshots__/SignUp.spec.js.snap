// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp Should match snapshot. 1`] = `
<div>
  <h1>
    Sign Up
  </h1>
   
  <form>
    <vuecomponent-stub
      name="first_name"
      placeholder="Joseph"
    />
     
    <div
      style="display: none;"
    >
      <validationmessage-stub />
       
      <!---->
    </div>
     
    <vuecomponent-stub
      name="last_name"
      placeholder="Bloggs"
    />
     
    <div
      style="display: none;"
    >
      <validationmessage-stub />
       
      <!---->
    </div>
     
    <vuecomponent-stub
      name="email"
      placeholder="joebloggs@example.com"
    />
     
    <div
      style="display: none;"
    >
      <validationmessage-stub />
       
      <!---->
    </div>
     
    <vuecomponent-stub
      name="username"
      placeholder="joe_bloggs"
    />
     
    <validationmessage-stub
      style="display: none;"
    />
     
    <vuecomponent-stub
      name="password"
      placeholder="Must have at least 5 characters."
      type="password"
    />
     
    <div
      style="display: none;"
    >
      <validationmessage-stub />
       
      <!---->
    </div>
     
    <vuecomponent-stub
      type="success"
    />
  </form>
</div>
`;

exports[`SignUp Should render a fullscreen spinner if the component is submitting. 1`] = `
<vuecomponent-stub
  fullscreen="true"
/>
`;
